### DISCUSSION TITLE
Using this to set the state is no possible

### DISCUSSION THREAD

**Discussion by avifatal** [pmndrs/valtio#544] [Q&A] [answered]
Hi,
I have the following code trying to set the state via `this`:
```ts
export class Store {
  public info = [{ status: "empty" }];
  public loadAll = () => {
    axios.get(url).then((x) => {
      this.info = x.data;
    });
  };
}
```

I can't make it work...
you can play around here: https://codesandbox.io/s/valtio-to-do-list-forked-70yhlx?file=/src/App.tsx:128-288

Thanks

**Comment by dai-shi** [comment, marked as answer]
Unfortunately, you seem to have some misconceptions with `this` behavior.

> https://github.com/pmndrs/valtio#react-via-usesnapshot
> Use of `this` is for expert users.
> Valtio tries best to handle `this` behavior but it's hard to understand without familiarity.

Here's the fix, but I'd recommend to avoid using `this`.
https://codesandbox.io/s/valtio-to-do-list-forked-7k7bn0?file=/src/App.tsx

---

So, my recommendation is the following instead:
```js
export const state = proxy({
  info: [{ status: "empty" }],
  loadAll: () => {
    axios.get(url).then((x) => {
      state.info = x.data;
    });
  },
});
```

**Reply by avifatal** [reply]
Thank you. can you please explain why using `this` is not recommended? 
I got the idea from the docs.
Thanks

**Reply by dai-shi** [reply]
> why using `this` is not recommended?

becuase it's hard to understand without familiarity.

**Reply by avifatal** [reply]
Thank you.
I wanted to use the `this` in order to have a base class and reuse async status `pending`, 'done', `error` and so...
I find it annoing to re-do it again and again...
Thanks
